### PR TITLE
Feature/add webworkers api widl

### PR DESCRIPTION
**Pull request by a-tarasyuk**
[Microsoft/TypeScript#/23653](https://github.com/Microsoft/TypeScript/issues/23653)
[Web Workers Spec](https://html.spec.whatwg.org/multipage/workers.html)

### CODE REVIEW

File: baselines/dom.generated.d.ts [hunk 1]
```diff
@@ -16632,5 +16657,5 @@ type VRDisplayEventReason = "mounted" | "navigation" | "requested" | "unmounted"
 type VREye = "left" | "right";
 type VideoFacingModeEnum = "user" | "environment" | "left" | "right";
 type VisibilityState = "hidden" | "visible" | "prerender" | "unloaded";
-type WorkerType = "classic" | "module";
+type WorkerType = "classic" | "module" | "classic" | "module";
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: this needs to be filtered.

File: baselines/webworker.generated.d.ts [hunk 2]
```diff
@@ -2067,42 +2120,14 @@ interface NotificationPermissionCallback {
...
-declare function msWriteProfilerMark(profilerMarkName: string): void;
-declare function dispatchEvent(evt: Event): boolean;
-declare var indexedDB: IDBFactory;
-declare var msIndexedDB: IDBFactory;
-declare var navigator: WorkerNavigator;
-declare function clearImmediate(handle: number): void;
-declare function clearInterval(handle: number): void;
-declare function clearTimeout(handle: number): void;
-declare function importScripts(...urls: string[]): void;
-declare function setImmediate(handler: any, ...args: any[]): number;
```
Comment (mhegazy) [inline baselines/webworker.generated.d.ts]: why are these removed?

File: baselines/dom.generated.d.ts (hunk 1 above)
Comment (a-tarasyuk) [reply baselines/dom.generated.d.ts]: done

File: baselines/webworker.generated.d.ts (hunk 2 above)
Comment (saschanaz) [reply baselines/webworker.generated.d.ts]: Because widlprocess does not support `[Global]` yet 😅

**Comment by saschanaz** [comment]
I was going to add full HTML spec in once, but being incremental is probably better 😁

File: baselines/webworker.generated.d.ts (hunk 2 above)
Comment (saschanaz) [reply baselines/webworker.generated.d.ts]: #456 is merged, please rebuild!

**Comment by saschanaz** [comment]
The multipaged worker spec lacks `WindowOrWorkerGlobalScope` which is the target from many specs to add global methods/properties. We may manually add an empty one.

**Comment by saschanaz** [comment]
Oh, this is an oops situation. I expected the empty WindowOrWorkerGlobalScope would automatically get some types from already added specs, but:

1. We don't support `partial interface mixin`s yet
2. Even if we did, specs are incorrectly using it as `partial interface`. I'll file some issue for those specs.

**Comment by a-tarasyuk** [comment]
@saschanaz should I remove it for now?

**Comment by saschanaz** [comment]
It also has IndexedDB and other stable types. Worker will lose access for them without proper WindowOrWorkerGlobalScope definition. I think we should add those lost types manually for now.

**Comment by a-tarasyuk** [comment]
@saschanaz Will add.

File: inputfiles/knownWorkerTypes.json [hunk 3]
```diff
@@ -1,6 +1,126 @@
 [
+    "ClientQueryOptions",
```
[the remark this answers is not in the document]
Comment (saschanaz) [reply inputfiles/knownWorkerTypes.json]: I meant not only that one but all additions here, because now types are autoexposed.

File: baselines/webworker.generated.d.ts [hunk 4]
```diff
@@ -2055,38 +2078,33 @@ interface EventHandlerNonNull {
...
-declare var indexedDB: IDBFactory;
-declare var msIndexedDB: IDBFactory;
-declare var navigator: WorkerNavigator;
-declare function clearImmediate(handle: number): void;
-declare function clearInterval(handle: number): void;
-declare function clearTimeout(handle: number): void;
 declare function importScripts(...urls: string[]): void;
-declare function setImmediate(handler: any, ...args: any[]): number;
-declare function setInterval(handler: any, timeout?: any, ...args: any[]): number;
-declare function setTimeout(handler: any, timeout?: any, ...args: any[]): number;
```
Comment (mhegazy) [inline baselines/webworker.generated.d.ts]: is that expected? should `setTimeout` not be available for workers?

Comment (saschanaz) [reply baselines/webworker.generated.d.ts]: It's needed, I think we should wait until we get `WindowOrWorkerGlobalScope`.
